src/MMALSharp.Processing/Handlers/ExternalProcessCaptureHandler.cs

```diff
@@ -0,0 +1,266 @@
+        /// Writes frame data to the StandardInput stream for processing.
+        /// </summary>
+        /// <param name="context">Contains the data and metadata for an image frame.</param>
+        public void Process(ImageContext context)
+        {
+            try
+            {
+                _process.StandardInput.BaseStream.Write(context.Data, 0, context.Data.Length);
+                _process.StandardInput.BaseStream.Flush();
+                this.Processed += context.Data.Length;
+            }
+            catch
+            {
+                Dispose();
+                throw;
+            }
+        }
+
+        /// <summary>
+        /// Not used.
+        /// </summary>
+        /// <exception cref="NotImplementedException"></exception>
+        public void Split()
+            => throw new NotImplementedException();
+
+        /// <summary>
+        /// Returns the total number of bytes processed by this capture handler.
+        /// </summary>
+        /// <returns>The total number of bytes processed by this capture handler.</returns>
+        public string TotalProcessed()
+            => $"{this.Processed}";
+
+        /// <summary>
+        /// Manages echoing the output buffer and handles attempts to cleanly terminate the
+        /// child process. Use the same CancellationToken passed to MMALCamera.ProcessAsync
+        /// and execute both of these with Task.WhenAll.
+        /// </summary>
+        /// <param name="cancellationToken">The same timeout token used for MMALCamera.ProcessAsync</param>
+        public async Task ManageProcessLifecycleAsync(CancellationToken cancellationToken)
+        {
+            var outputToken = new CancellationTokenSource();
+
+            await Task.WhenAny(new[]
+            {
+                // this Task is the one that will be cancelled by the ProcessAsync timeout
+                cancellationToken.AsTask(),
+
+                // we control this token so this will keep running when the above expires
+                ConsoleWriteLineAsync(outputToken.Token)
+            }).ConfigureAwait(false);
+
+            // now we can do a clean shutdown; ConsoleWriteLineAsync is still running
+            if (_options.TerminationSignals.Length > 0)
+            {
+                MMALLog.Logger.LogTrace($"Sending process termination signals");
+                foreach (var sigint in _options.TerminationSignals)
+                {
+                    if (_process.HasExited)
+                    {
+                        break;
+                    }
+
+                    Syscall.kill(_process.Id, sigint);
+                }
+            }
+
+            // gives output buffer a chance to drain, and also allows the process to do exit-cleanup
+            var stopwatch = new Stopwatch();
+            stopwatch.Start();
+            while (!_process.HasExited && stopwatch.ElapsedMilliseconds < _options.DrainOutputDelayMs)
+            {
+                await Task.Delay(50);
```

Comment:
Thanks Jon for your updates and well spotted about `ConfigureAwait`. These `await` calls will also need it adding too please.